ethr-status-registry/src/main/java/me.uport.sdk.ethrstatusregistry/EthrStatusRegistry.kt

```diff
@@ -0,0 +1,88 @@
+package me.uport.sdk.ethrstatusregistry
+
+import kotlinx.coroutines.runBlocking
+import me.uport.credential_status.CredentialStatus
+import me.uport.credential_status.StatusEntry
+import me.uport.credential_status.StatusResolver
+import me.uport.sdk.core.Networks
+import me.uport.sdk.jsonrpc.JsonRPC
+import me.uport.sdk.jwt.JWTTools
+import org.kethereum.extensions.hexToBigInteger
+import org.kethereum.keccakshortcut.keccak
+import pm.gnosis.model.Solidity
+
+class EthrStatusRegistry : StatusResolver {
+
+    override val method = "EthrStatusRegistry2019"
+
+    override fun checkStatus(credential: String): CredentialStatus {
+        val (_, payloadRaw) = JWTTools().decodeRaw(credential)
+        val status = payloadRaw["status"] as Map<String, String>
+        val type = status["type"] ?: ""
+        val id = status["id"] ?: ""
+        val statusEntry = StatusEntry(type, id)
+
+        if (statusEntry.type == method) {
+            return runCredentialCheck(
+                credential,
+                statusEntry
+            )
+        } else {
+            throw IllegalStateException("The method '$method' is not a supported credential status method.")
+        }
+    }
+
+    private fun runCredentialCheck(
+        credential: String,
+        status: StatusEntry
+    ): CredentialStatus {
+        val (address, network) = parseRegistryId(status.id)
+
+        val ethNetwork = Networks.get(network)
+        val rpc = JsonRPC(ethNetwork.rpcUrl)
+        val credentialHash = credential.toByteArray().keccak()
+
+        val encodedMethodCall = Revocation.Revoked.encode(
+            Solidity.Address(address.hexToBigInteger()),
+            Solidity.Bytes32(credentialHash)
+        )
+
+        val result = runBlocking {
+            rpc.ethCall(address, encodedMethodCall)
+        }
+
+        if (result.toBigIntegerOrNull() != null) {
+            return CredentialStatus(true)
+        } else {
+            return CredentialStatus(false)
+        }
+    }
+
+    private fun parseRegistryId(did: String): Pair<String, String> {
+
+        //language=RegExp
+        val didParsePattern = "^(did:)?((\\w+):)?((\\w+):)?((0x)([0-9a-fA-F]{40}))".toRegex()
```

Comment:
The `status.id` field is not a DID; it has the following format: `networkName:0xaddress`
Example: `mainnet:0x63142bebe28e663ec4d007cd12ce5e77c37deafc`

This means that the regex can be simplified